src/node_url.h

```diff
@@ -3,201 +3,28 @@
-};
-
-enum url_flags {
-#define XX(name, val) name = val,
-  FLAGS(XX)
-#undef XX
-};
-
-struct url_data {
-  int32_t flags = URL_FLAGS_NONE;
-  int port = -1;
-  std::string scheme;
-  std::string username;
-  std::string password;
-  std::string host;
-  std::string query;
-  std::string fragment;
-  std::vector<std::string> path;
-  std::string href;
-};
-
-namespace table_data {
-extern const char hex[1024];
-extern const uint8_t C0_CONTROL_ENCODE_SET[32];
-extern const uint8_t FRAGMENT_ENCODE_SET[32];
-extern const uint8_t PATH_ENCODE_SET[32];
-extern const uint8_t USERINFO_ENCODE_SET[32];
-extern const uint8_t QUERY_ENCODE_SET_NONSPECIAL[32];
-extern const uint8_t QUERY_ENCODE_SET_SPECIAL[32];
-}
-
-class URL {
- public:
-  static void Parse(const char* input,
-                    size_t len,
-                    enum url_parse_state state_override,
-                    struct url_data* url,
-                    bool has_url,
-                    const struct url_data* base,
-                    bool has_base);
-
-  static std::string SerializeURL(const url_data& url, bool exclude);
-
-  URL(const char* input, const size_t len) {
-    Parse(input, len, kUnknownState, &context_, false, nullptr, false);
-  }
-
-  URL(const char* input, const size_t len, const URL* base) {
-    if (base != nullptr)
-      Parse(input, len, kUnknownState,
-            &context_, false,
-            &(base->context_), true);
-    else
-      Parse(input, len, kUnknownState, &context_, false, nullptr, false);
-  }
-
-  URL(const char* input, const size_t len,
-      const char* base, const size_t baselen) {
-    if (base != nullptr && baselen > 0) {
-      URL _base(base, baselen);
-      Parse(input, len, kUnknownState,
-            &context_, false,
-            &(_base.context_), true);
-    } else {
-      Parse(input, len, kUnknownState, &context_, false, nullptr, false);
-    }
-  }
-
-  explicit URL(const std::string& input) :
-      URL(input.c_str(), input.length()) {}
-
-  URL(const std::string& input, const URL* base) :
-      URL(input.c_str(), input.length(), base) {}
-
-  URL(const std::string& input, const URL& base) :
-      URL(input.c_str(), input.length(), &base) {}
-
-  URL(const std::string& input, const std::string& base) :
-      URL(input.c_str(), input.length(), base.c_str(), base.length()) {}
-
-  int32_t flags() const {
-    return context_.flags;
-  }
-
-  int port() const {
-    return context_.port;
-  }
-
-  const std::string& protocol() const {
-    return context_.scheme;
-  }
-
-  const std::string& username() const {
-    return context_.username;
-  }
-
-  const std::string& password() const {
-    return context_.password;
-  }
-
-  const std::string& host() const {
-    return context_.host;
-  }
-
-  const std::string& query() const {
-    return context_.query;
-  }
-
-  const std::string& fragment() const {
-    return context_.fragment;
+static inline std::string FromFilePath(const std::string_view file_path) {
```

Comment:
See earlier conversation: https://github.com/nodejs/node/pull/46410#discussion_r1096095567

I meant, why define the function in the header rather than in the .cc file? I.e., I don't think it should be inlined.

Reply:
Updating the pull request. Thanks!